src/header/internal.tsx

```diff
@@ -60,10 +62,20 @@ export default function InternalHeader({
             isRefresh && styles[`title-variant-${variantOverride}-refresh`]
           )}
         >
-          <HeadingTag className={clsx(styles.heading, styles[`heading-variant-${variantOverride}`])}>
-            <span className={clsx(styles['heading-text'], styles[`heading-text-variant-${variantOverride}`])}>
+          <HeadingTag
+            className={clsx(
+              styles.heading,
+              styles[`heading-variant-${variantOverride}`],
+              truncateHeading && styles['heading-overflow']
+            )}
+          >
+            <span
+              className={clsx(styles['heading-text'], styles[`heading-text-variant-${variantOverride}`])}
+              aria-hidden={truncateHeading}
+            >
               {children}
             </span>
+            {truncateHeading && <ScreenreaderOnly>{children}</ScreenreaderOnly>}
```

Comment:
Why is this needed, what is wrong with the visible text?

`ScreenreaderOnly` component works as maximum truncation down to 1px, so what is wrong with relying on half-truncated content?

Reply:
I just bumped into this as well when I noticed that there are two DOM nodes with the same text when aria-hidden='true'.

<img width="446" alt="Screen Shot 2023-02-13 at 4 05 34 PM" src="https://user-images.githubusercontent.com/438181/218574360-1dd5f54b-f478-48db-90d1-38ac88d6bcb9.png">

I don't think I understand what problem is solved by putting aria-hidden on the truncated node and then replicating the same text and rendering it off the canvas...

Reply:
That allows screen-reader users to always read the full contents no matter if the actual header is truncated or not.

I don't mind removing it for now and adding back in case we'll have the feedback.

Reply:
Pardon my ignorance but isn't the opposite true? As long as aria hidden is not set, the presence of hidden overflow will still announce the entire content to a screenreader?